impl.go

```diff
@@ -246,6 +251,28 @@ func (channel *channelImpl) Send(s Sendable) error {
 	return nil
 }
 
+func (channel *channelImpl) TrySend(s Sendable) (bool, error) {
+	if err := s.Context().Err(); err != nil {
+		return false, err
+	}
+
+	s.SetSequence(int32(channel.sequence.Next()))
+
+	select {
+	case <-s.Context().Done():
+		if err := s.Context().Err(); err != nil {
+			return false, TimeoutError{errors.Wrap(err, "timeout waiting for message reply")}
+		}
+		return false, errors.New("timeout waiting to put message in send queue")
+	case <-channel.closeNotify:
+		return false, errors.New("channel closed")
+	case channel.outQueue <- s:
+		return true, nil
+	default:
+		return false, nil
```

Comment:
should this also return an error (like the timeout case)?

Reply:
this maybe a case for Typed/Const errors to make it easier on the caller

Reply:
I added a ClosedError to make it easier to distinguish those.

I think a return of `false, nil` indicates that the queue was full. If we want an error type for that, we don't need the boolean.